state/enableha.go

```diff
@@ -280,7 +289,6 @@ func (st *State) enableHAIntentions(info *ControllerInfo, placement []string) (*
 				return nil, errors.Errorf("machine for placement directive %q is already a controller", s)
 			}
 			intent.convert = append(intent.convert, m)
-			intent.placement = append(intent.placement, s)
```

Comment:
This is the critical change. 

If we use a placement directive to target an existing machine, don't add it to the generated intentions - those are only used when adding new controllers.